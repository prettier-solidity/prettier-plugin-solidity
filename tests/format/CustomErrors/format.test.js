runFormatTest(import.meta, ['solidity-parse']);
