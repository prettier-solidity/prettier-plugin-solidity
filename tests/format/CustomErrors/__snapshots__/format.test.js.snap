// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomErrors.sol format 1`] = `
====================================options=====================================
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.8.4;

error
  TopLevelCustomError();
  error TopLevelCustomErrorWithArg(uint    x)  ;
error TopLevelCustomErrorArgWithoutName  (string);

contract CustomErrors {
  error
    ContractCustomError();
    error ContractCustomErrorWithArg(uint    x)  ;
  error ContractCustomErrorArgWithoutName  (string);

  function throwCustomError() {
    revert
      FunctionCustomError();
    revert FunctionCustomErrorWithArg(  1  )  ;
    revert    FunctionCustomErrorArgWithoutName  (  "a reason");
  }
}

=====================================output=====================================
pragma solidity ^0.8.4;

error TopLevelCustomError();
error TopLevelCustomErrorWithArg(uint x);
error TopLevelCustomErrorArgWithoutName(string);

contract CustomErrors {
    error ContractCustomError();
    error ContractCustomErrorWithArg(uint x);
    error ContractCustomErrorArgWithoutName(string);

    function throwCustomError() {
        revert FunctionCustomError();
        revert FunctionCustomErrorWithArg(1);
        revert FunctionCustomErrorArgWithoutName("a reason");
    }
}

================================================================================
`;
